reconcile/utils/terraform_client.py

```diff
@@ -823,6 +823,26 @@ def validate_db_upgrade(
                     changed_fields=changed_fields,
                 )
 
+            response = self._aws_api.describe_rds_db_instance(
+                account_name, resource_name, region_name=region_name
+            )
+            replica_identifiers = response["DBInstances"][0].get(
+                "ReadReplicaDBInstanceIdentifiers"
+            )
+            if replica_identifiers is not None:
+                if len(replica_identifiers) > 0:
+                    replica_db = replica_identifiers[0]
+                    replica_db_response = self._aws_api.describe_rds_db_instance(
+                        account_name, replica_db, region_name=region_name
+                    )
+                    replica_version = replica_db_response["DBInstances"][0].get(
+                        "EngineVersion"
+                    )
+                    if after.get("engine_version") > replica_version:
+                        raise RdsUpgradeValidationError(
+                            "The replica database is running a previous version from the primary and needs to have an upgrade first."
```

Comment:
Re-reading our docummentation, we recommend to remove replicas before upgrade... shouldn't be this message more geared towards that fact? https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/dba/postgresql-rds-instance-major-version-upgrade.md#note-about-upgrading-read-replicas

I'm worried that people actually tries to do what this message says rather than what our documentation says. Just thinking out loud here 😅

Reply:
It's a good point! If we think it's better to error out on recommendations we give to tenants, I'm on board with focusing on that. Is the deletion of the read replica specific to only postgres? Or should we keep the rule general for all engine types?

Reply:
Everybody uses postgres... we support MySQL because we use it ourselves in Vault commercial database, but the contract says Postgres :)

Reply:
Okie dokie! I'll put this logic on hold and change it to reflect the replica removal logic check

Reply:
I'm not sure what you have in mind. I was just meaning something simple about the messaging. Something like "Detected upgrade of a Postgres RDS database with a read-replica. This is not supported by AWS. Please delete the replica before proceeding"

Reply:
That's what I had in mind, yes